### PR TITLE
Feature/l2 ethereum snapshot

### PR DESCRIPTION
**Pull request by jeremyletang**

### CODE REVIEW

File: core/client/eth/l2_clients.go [hunk 1]
```diff
@@ -113,50 +113,50 @@ func (e *L2Clients) ReloadConf(cfg Config) {
 
 	e.log.Info("updating L2 clients")
 	for _, v := range cfg.L2Configs {
-		if _, ok := e.clients[v.NetworkID]; ok {
+		if _, ok := e.clients[v.ChainID]; ok {
 			e.log.Warn("L2 client already setted up, please stop and restart node to update existing configuration",
-				logging.String("network-id", v.NetworkID),
+				logging.String("network-id", v.ChainID),
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:118]: missed one

File: core/client/eth/l2_clients.go [hunk 2]
```diff
@@ -113,50 +113,50 @@ func (e *L2Clients) ReloadConf(cfg Config) {
...
 				logging.String("endpoint", v.RPCEndpoint),
 			)
 			continue
 		}
 
 		e.log.Info("starting L2 client",
-			logging.String("network-id", v.NetworkID),
+			logging.String("network-id", v.ChainID),
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:125]: missed 2

File: core/client/eth/l2_clients.go [hunk 3]
```diff
@@ -113,50 +113,50 @@ func (e *L2Clients) ReloadConf(cfg Config) {
...
 			logging.String("endpoint", v.RPCEndpoint),
 		)
-		if len(v.NetworkID) <= 0 || len(v.RPCEndpoint) <= 0 {
+		if len(v.ChainID) <= 0 || len(v.RPCEndpoint) <= 0 {
 			e.log.Warn("invalid L2 client configuration",
-				logging.String("network-id", v.NetworkID),
+				logging.String("network-id", v.ChainID),
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:130]: 3

File: core/client/eth/l2_clients.go [hunk 4]
```diff
@@ -113,50 +113,50 @@ func (e *L2Clients) ReloadConf(cfg Config) {
...
 				logging.String("endpoint", v.RPCEndpoint),
 			)
 			continue
 		}
 		clt, err := DialL2(e.ctx, v.RPCEndpoint)
 		if err != nil {
 			e.log.Warn("couldn't start L2 client",
-				logging.String("network-id", v.NetworkID),
+				logging.String("network-id", v.ChainID),
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:138]: 4

File: core/client/eth/l2_clients.go [hunk 5]
```diff
@@ -113,50 +113,50 @@ func (e *L2Clients) ReloadConf(cfg Config) {
...
 				logging.String("endpoint", v.RPCEndpoint),
 			)
 			continue
 		}
 
-		networkID, err := clt.NetworkID(e.ctx)
+		networkID, err := clt.ChainID(e.ctx)
 		if err != nil {
 			e.log.Warn("couldn't get network id", logging.Error(err))
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:146]: 6

File: core/client/eth/l2_clients.go [hunk 6]
```diff
@@ -113,50 +113,50 @@ func (e *L2Clients) ReloadConf(cfg Config) {
...
 			continue
 		}
 
-		if networkID.String() != v.NetworkID {
+		if networkID.String() != v.ChainID {
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:150]: 7

**Review by candida-d — APPROVED** [review]

File: core/client/eth/l2_clients.go [hunk 7]
```diff
@@ -54,28 +54,28 @@ func NewL2Clients(
...
-		networkID, err := clt.NetworkID(ctx)
+		networkID, err := clt.ChainID(ctx)
 		if err != nil {
 			return nil, fmt.Errorf("couldn't get network id: %w", err)
 		}
 
-		if networkID.String() != v.NetworkID {
-			return nil, fmt.Errorf("client retrieve different network id: %v vs %v", networkID.String(), v.NetworkID)
+		if networkID.String() != v.ChainID {
+			return nil, fmt.Errorf("client retrieve different network id: %v vs %v", networkID.String(), v.ChainID)
```
Comment (EVODelavega) [inline core/client/eth/l2_clients.go:74]: missed one again

**Review by EVODelavega — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**